rewrite-test/src/main/java/org/openrewrite/test/RecipeSpec.java

```diff
@@ -19,7 +19,6 @@
 import org.openrewrite.config.Environment;
```

Comment:
I think what we really need here is a way to optionally contribute a `MavenSettings` to the runTest. Where RewriteTest has something like `mavenSettings(String)` and `mavenSettings(MavenSettings)`. I hesitate to add something that reads from local disk, as this could result in very different behavior of the unit tests between machines. I think that if an organization or user want to use the local maven settings, they really need to opt-in to that behavior by extending our base testing infrastructure. Does that make sense?

Reply:
hmmmmmm....

Yeah, I hear you. Making a large swath of unit tests dependent on external files on the executing machine sounds fundamentally like an anti-pattern. And increasing extensibility of `RewriteTest` for this use case sounds like a good compromise.

I do think the idea should get shelved and not discarded, though. A few observations:
* As written up in #1801, this *feels* like a defect, so we might want to give any incremental solution a place of honor in documentation, especially if we recommend extending your own `MyRewriteTest` interface as a pattern
* This behavior is pesky for folks like me who fork-build-contribute w/ OpenRewrite within one of those walled environments, as all relevant tests won't have access to my settings.xml. I either draft code changes which won't get committed (disabling or supplying execution context) for those tests, or live w/ having some doomed-to-fail tests in my local build

Regardless, for increasing extensibility:
My first thought was adding `defaultExecutionContext()` as an interface method on `RewriteTest`, but `mavenSettings(...)` makes sense too. I'd just want any solution to be composition-friendly: as an anti-example, if I extended `RewriteTest` in my personal project and provided a new implementation for `defaults` to add this config, then my test classes would need to remember to call the `super` if they also override `defaults`, which is different from working with `RewriteTest` directly and likely to introduce error.